### ISSUE TITLE
Unbound variables not detected

### ISSUE THREAD

**Issue by erictraut** [Microsoft/pyright#104]
In a recent rewrite / simplification of the type checking logic, I eliminated the code that detects the use of unbound or potentially-unbound variables. This needs to be added back.

**Comment by tomisaacson** [comment]
Is this why on a simple example:
```
import boto3
sts = boto3.resource('sts')
```
I'm getting:
* 'boto3' is not bound
* Cannot access member 'resource' on type 'Unbound'

**Comment by erictraut** [comment]
@tomisaacson, what you're seeing is a bad regression that slipped through my testing. It occurs when the `import xyz` statement is used to import a module that has no type stub file. In our internal projects where we use pyright, we have type stub files defined for all of our externally-imported modules, so I didn't notice this bug. I'll need to augment the test automation to detect similar problems in the future. 

I've done a quick-turn publish of version 1.0.22 which should eliminate the problem you're seeing.

**Comment by tomisaacson** [comment]
Working now, thanks.

**Comment by erictraut** [comment]
Unbound variables are now being properly detected again as of 1.0.24

**Comment by Jbiloki** [comment]
This is not working correctly again on the current version, importing requests is still unbound.

**Comment by florianschleuss** [comment]
> This is not working correctly again on the current version, importing requests is still unbound.

Got same issue. Works for all other imports but requests module is "unbound"

**Comment by erictraut** [comment]
@florianschleuss, please open a new issue. This one was fixed, so it sounds like you're seeing something new.
